src/main/java/org/gridsuite/modification/server/modifications/TwoWindingsTransformerModification.java

```diff
@@ -438,29 +459,88 @@ private void addPhaseTapChangersToTwoWindingsTransformer(Network network, TwoWin
-            if (!phaseTapChanger.isRegulating()) {
-                regulationMode = PhaseTapChanger.RegulationMode.FIXED_TAP;
-            }
+            regulationMode = oldRegulationMode;
         }
         if (!PhaseTapChanger.RegulationMode.FIXED_TAP.equals(regulationMode)) {
             modifyPhaseTapRegulation(phaseTapChangerInfos, phaseTapChanger, phaseTapChangerAdder, regulationReports, regulationMode);
         }
         modifyRegulatingTerminal(phaseTapChangerInfos, phaseTapChangerAdder, phaseTapChanger, regulationReports, network, twt.getTerminal1(), twt.getTerminal2());
-        if (!regulationReports.isEmpty()) {
+
+        List<Report> positionsAndStepsReports = new ArrayList<>();
+        addTapChangerPositionsAndSteps(phaseTapChangerInfos, phaseTapChanger, phaseTapChangerAdder, positionsAndStepsReports);
+
+        // TODO : This is a workaround that needs to be removed when modifying tap changer steps will be possible
+        // workaround : we need to remove the tap changer before adding the new one to avoid having two tap changers on the same transformer
+        // and we need to prepare a backup in case the addition fails
+        PhaseTapChangerAdder phaseTapChangerBackUp = null;
+        if (phaseTapChanger != null) {
+            phaseTapChangerBackUp = rollbackPhaseModificationsAdder(twt, phaseTapChanger);
+            phaseTapChanger.remove();
+        }
+
+        try {
+            phaseTapChangerAdder.add();
+        } catch (Exception e) {
+            if (phaseTapChangerBackUp != null) {
+                phaseTapChangerBackUp.add();
+            }
+            throw e;
+        }
+
+        if (!phaseTapChangerReports.isEmpty() || !regulationReports.isEmpty() || !positionsAndStepsReports.isEmpty()) {
+            Reporter phaseTapChangerSubreporter = ModificationUtils.getInstance().reportModifications(subReporter, phaseTapChangerReports, TapChangerType.PHASE.name(), PHASE_TAP_CHANGER_SUBREPORTER_DEFAULT_MESSAGE);
+            if (phaseTapChangerSubreporter == null) {
+                phaseTapChangerSubreporter = subReporter.createSubReporter(TapChangerType.PHASE.name(), PHASE_TAP_CHANGER_SUBREPORTER_DEFAULT_MESSAGE);
+                phaseTapChangerSubreporter.report(Report.builder()
+                            .withKey(TapChangerType.PHASE.name())
+                            .withDefaultMessage(PHASE_TAP_CHANGER_SUBREPORTER_DEFAULT_MESSAGE)
+                            .withSeverity(TypedValue.INFO_SEVERITY)
+                            .build());
+            }
             ModificationUtils.getInstance().reportModifications(phaseTapChangerSubreporter, regulationReports, regulationMode != null ? regulationMode.name() : null, ModificationUtils.getInstance().formatRegulationModeReport(regulationMode));
+            ModificationUtils.getInstance().reportModifications(phaseTapChangerSubreporter, positionsAndStepsReports, "phaseTapChangerPositionsAndStepsModification", "    Tap Changer");
         }
+    }
```

Comment:
could you specify the scope of the code to be removed ?
do we we have to remove everything from TODO to the end of this function ?